### PR TITLE
Add library.get_images()

### LINKED ISSUE #973: Implement core.library.get_images(uris)
See http://goo.gl/TOzY8L for details. Should be used to implement #809 and will fix #838

### PR DESCRIPTION
**Pull request by adamcik**
Fixes #973 and #838 - followup will be to start deprecating the old images in models.

### CODE REVIEW

File: mopidy/core/library.py [hunk 1]
```diff
@@ -72,6 +72,22 @@ def browse(self, uri):
             return []
         return backend.library.browse(uri).get()
 
+    def get_images(self, uris):
+        """Lookup the images for the given URIs
+
+        :param list uris: list of uris to find images for
+        :rtype: dict mapping uris to :class:`mopidy.models.Image` instances
```
Comment (jodal) [inline mopidy/core/library.py]: s/uris/URIs/ in the text.

Comment (jodal) [inline mopidy/core/library.py]: It isn't clear if the mapping is from URI to Image or URI to list of Image.

File: tests/test_models.py [hunk 2]
```diff
@@ -123,11 +123,30 @@ def test_playlist_constructor(self):
         self.assertEqual(ref.name, 'bar')
         self.assertEqual(ref.type, Ref.PLAYLIST)
 
-    def test_track_constructor(self):
-        ref = Ref.track(uri='foo', name='bar')
-        self.assertEqual(ref.uri, 'foo')
-        self.assertEqual(ref.name, 'bar')
-        self.assertEqual(ref.type, Ref.TRACK)
```
Comment (jodal) [inline tests/test_models.py]: Was removing this test intentional?

Comment (adamcik) [inline tests/test_models.py]: Nope, will fix.

File: mopidy/core/library.py (hunk 1 above)
Comment (adamcik) [inline mopidy/core/library.py]: I implemented it wrong, it should be `{uri: [images]}` - and yes to description can also be more clear.

File: tests/core/test_library.py [hunk 3]
```diff
@@ -33,6 +37,46 @@ def setUp(self):  # noqa: N802
...
             self.backend1, self.backend2, self.backend3])
 
+    def test_get_images_returns_empty_dict_for_no_uris(self):
+        self.assertEqual({}, self.core.library.get_images([]))
+
+    def test_get_images_returns_empty_dict_for_unknown_uri(self):
+        self.assertEqual({}, self.core.library.get_images(['dummy4:bar']))
+
+    def test_get_images_returns_empty_dict_for_library_less_uri(self):
+        self.assertEqual({}, self.core.library.get_images(['dummy3:foo']))
```
Comment (adamcik) [inline tests/core/test_library.py]: Should we return the empty dict, or should we return `{'dummy3:foo': []}`?

Comment (jodal) [inline tests/core/test_library.py]: Latter, I think. That way we only have one "failure result".

**Comment by adamcik** [comment]
There, that should be better.

File: tests/core/test_library.py [hunk 4]
```diff
@@ -33,6 +37,50 @@ def setUp(self):  # noqa: N802
...
+        self.library1.get_images.reset_mock()
+
+        result = self.core.library.get_images(['dummy1:track'])
+        self.assertEqual({'dummy1:track': (Image(uri='uri'),)}, result)
+
+    def test_get_images_merges_results(self):
+        self.library1.get_images().get.return_value = {
+            'dummy1:track': [Image(uri='uri1')]}
+        self.library1.get_images.reset_mock()
+        self.library2.get_images().get.return_value = {
```
Comment (jodal) [inline tests/core/test_library.py]: I know we've used this pattern some places in our tests, but the proper way to do this (without registering an extra call to `get_images()`) is to `s/get_images().get/get_images.return_value.get/`.

**Pull request merged**